processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorProperties.java

```diff
@@ -102,14 +102,26 @@
             PropertyDefinition.define("decaton.group.rebalance.timeout.ms", Long.class, 1000L,
                                       v -> v instanceof Long && (Long) v >= 0);
 
+    /**
+     * Timeout for processor close. Decaton waits up to this time for tasks currently in-progress to finish.
+     * Any tasks that do not complete within this timeout will mean async task processing code may still
+     * be running even after {@link ProcessorSubscription#close()} returns, which might lead to errors from e.g.
+     * shutting down dependencies of this {@link ProcessorSubscription} that are still in use from async tasks.
+     * Reloadable: yes
+     */
+    public static final PropertyDefinition<Long> CONFIG_CLOSE_TIMEOUT_MS =
+            PropertyDefinition.define("decaton.processing.close.timeout.ms", Long.class, 0L,
```

Comment:
I think the majority of users will call `close()` (directly or via a framework) and it will be more advanced users who call `initiateShutdown()` and `awaitShutdown()`. But I don't have a strong view about the name, happy to use `shutdown` if you think that's clearer.

Reply:
Ok, let's go with `shutdown` then.